include/hcl/base/containers/concurrent_skiplist/SkiplistAccessor.h

```diff
@@ -0,0 +1,323 @@
+#ifndef HCL_SKIPLIST_ACCESSOR_H
+#define HCL_SKIPLIST_ACCESSOR_H
+
+template <typename T, typename Comp, typename NodeAlloc, int MAX_HEIGHT>
+class ConcurrentSkipList<T, Comp, NodeAlloc, MAX_HEIGHT>::Accessor 
```

Comment:
if this is the class Maybe filename should be concurrent_skip_list?

Also for organization have it as concurrent/skip_list/skip_list_inl<full>.h
concurrent/skip_list/skip_list.h
concurrent/skip_list/skip_list_accessor.h